tests/functional/lib/helpers.js

```diff
@@ -696,45 +696,46 @@ define([
     var age = options.age || 24;
     var submit = options.submit !== false;
 
-    var remote = getRemote(context);
-    return remote
-      .getCurrentUrl()
-      .then(function (currentUrl) {
-        // only load the signup page if not already at a signup page.
-        // the leading [\/#] allows for either the standard redirect or iframe
-        // flow. The iframe flow must use the window hash for routing.
-        if (! /[\/#]signup(?:$|\?)/.test(currentUrl)) {
-          return remote
-            .get(require.toUrl(SIGNUP_URL))
-            .setFindTimeout(intern.config.pageLoadTimeout);
-        }
-      })
+    return function () {
+      return this.parent
+        .getCurrentUrl()
+        .then(function (currentUrl) {
+          // only load the signup page if not already at a signup page.
+          // the leading [\/#] allows for either the standard redirect or iframe
+          // flow. The iframe flow must use the window hash for routing.
+          if (! /[\/#]signup(?:$|\?)/.test(currentUrl)) {
+            return this.parent
```

Comment:
As in the other PR, I believe this `this` won't point to the correct object, since a normal function is passed to `then`.